rocAL/rocAL_pybind/amd/rocal/plugin/pytorch.py

```diff
@@ -194,14 +130,14 @@ def __next__(self):
         else:
             if(self.loader._oneHotEncoding == True):
                 self.loader.GetOneHotEncodedLabels(self.labels, self.device)
-                self.labels_tensor = self.labels.view(-1, self.bs, self.loader._numOfClasses).long()
+                self.labels_tensor = self.labels.reshape(-1, self.bs, self.loader._numOfClasses)
```

Comment:
.view uses the same underlying buffer and returns a modified view of the same tensor so when we do .long(), it will be change the self.labels dtype so the next time you use the tensor for getting labels, it will be of wrong dtype